src/ldp/representation/RepresentationMetadata.ts

```diff
@@ -1,47 +1,219 @@
-/**
- * Contains metadata relevant to a representation.
- */
-import type { Quad } from 'rdf-js';
+import { quad as createQuad, literal, namedNode } from '@rdfjs/data-model';
+import { Store } from 'n3';
+import type { BlankNode, Literal, NamedNode, Quad, Term } from 'rdf-js';
+import { CONTENT_TYPE } from '../../util/MetadataTypes';
+
+const shortcuts: { [id: string]: NamedNode } = {
+  contentType: CONTENT_TYPE,
+};
 
 /**
- * Metadata corresponding to a {@link Representation}.
+ * Stores the metadata triples and provides methods for easy access.
+ * Most functions return the metadata object to allow for chaining.
  */
-export interface RepresentationMetadata {
+export class RepresentationMetadata {
+  private store: Store;
+  private id: NamedNode | BlankNode;
+
+  /**
+   * @param identifier - Identifier of the resource relevant to this metadata.
+   *                     A blank node will be generated if none is provided.
+   *                     Strings will be converted to named nodes. @ignored
+   *
+   * `@ignored` tag is necessary for Components-Generator.js
+   */
+  public constructor(identifier?: NamedNode | BlankNode | string);
+
+  /**
+   * @param metadata - Starts as a copy of the input metadata.
+   */
+  public constructor(metadata?: RepresentationMetadata);
+
+  public constructor(input?: NamedNode | BlankNode | string | RepresentationMetadata) {
+    this.store = new Store();
+    if (!input) {
+      this.id = this.store.createBlankNode();
+    } else if (typeof input === 'string') {
+      this.id = namedNode(input);
+    } else if (!(input instanceof RepresentationMetadata)) {
+      this.id = input;
+    } else {
+      this.id = input.identifier;
+      this.addQuads(input.quads());
+    }
+  }
+
+  /**
+   * @returns All metadata quads.
+   */
+  public quads(): Quad[] {
+    return this.store.getQuads(null, null, null, null);
+  }
+
+  /**
+   * Identifier of the resource this metadata is relevant to.
+   * Will update all relevant triples if this value gets changed.
+   */
+  public get identifier(): NamedNode | BlankNode {
+    return this.id;
+  }
+
+  public set identifier(id: NamedNode | BlankNode) {
+    const quads = this.quads().map((quad): Quad => this.convertQuad(this.id, id, quad));
```

Comment:
Might want to change the private method into an independent function, so you don't need to create the closure here.

Reply:
Not sure what you mean exactly

Reply:
`convertQuad` is now a private member function of the class, but you can just make it non-member function since it does not use `this`.